designs/embed_eksd_manifest.md

```diff
@@ -0,0 +1,90 @@
+# Embed EKS-D resources in EKS-A cluster
+
+## Introduction
+
+**Problem:** Currently, the EKS Anywhere controller pulls the EKS Distro manifest from the release URL provided in the versions bundle, which requires access to the internet.
+It would be ideal to have this manifest and its CRDs embedded in the cluster to allow the controller to operate without the requirement to make that external call.
+This will eventually be needed for air-gapped support.
+
+### Goals and Objectives
+
+As an EKS Anywhere user:
+
+* I want the EKSA controller to be able to operate without internet access
+* I want EKSD components to be supported during upgrade
+
+## Overview of Solution
+
+With this feature, the EKSD manifest & its CRD will be applied directly to the cluster.
+This will allow the controller to grab the EKSD release resources from within the cluster instead of reading in the EKSD manifest from the internet when fetching the applied spec.
+We will do this similarly to how we apply the EKSA components to the cluster, introducing a new task to the create workflow called `InstallEksdComponentsTask`.
+In the upgrade workflow, we will also add a step to apply the EKSD resources as we update other cluster resources.
+
+#### EKS-D bundle changes
+
+We currently don’t pull in the EKSD release CRD in the bundle, but we need to apply this CRD before applying the release manifest.
+I propose introducing a new field `components` in the EKSD bundle to refer to the file in S3 that stores this CRD.
+
+While most of our bundles have a `version` field, the EKS-D bundle does not have this field.
+Instead, the version is typically decided by the `releaseChannel` and the `releaseNumber`.
```

Comment:
I think here `name` is our `version`
It uniquely identifies the eksd `Release` object. And it will also be the name of the object in the cluster once we do the "apply"
Can we use that instead of using version channel and number? In fact, `name` contains both the release channel and release number.

Reply:
Yes from the way that we construct the `name`, we could use that as our version here. The only reason why I'm thinking to use release channel & number instead is to depend on the true source of those values rather than what we construct for the name. Thinking of a scenario where the name is constructed differently, but it's definitely an edge case.

Reply:
That name should be the same name that we find in `metadata.name` in the eks-d manifest, right?

Reply:
Yes that’s the same name

Reply:
Cool, then my vote would be to rely on this name